Pass-finder scripts supports argument, moved yklua test suite to same folder as pass-finder script.

nmdis1999 opened the pull request:
This PR addresses two changes:
1) Earlier pass-finder script stripped off arguments from passes, with changes from this commit we will now be able to use passes with arguments.
2) Moved yklua test suite to same folder as pass-finder script.

vext01 commented on ykrt/try_passes.py:
```diff
@@ -62,10 +59,10 @@ def get_all_passes(is_prelink):
+        if descr not in seen:
+            seen.add(descr)
+            passes.append(Pass(descr))
+    
```
So we simply pass through the default arguments to each pass?

It wasn't what I expected, but it's probably a decent first step.

(I thought that we might try different sets of arguments, but let's not do that right now)

nmdis1999 replied:
oh, i have a question: where do we get these new arguments from if not from default?

vext01 commented on ykrt/try_passes.py:
```diff
@@ -83,7 +80,7 @@ def test_pipeline(logf, pl):
     env["LINKTIME_PASSES"] = ",".join([p.name for p in pl.link_time])
 
-    p = Popen("try_repeat 10 sh test.sh 2>&1", cwd=CWD, shell=True,
+    p = Popen("try_repeat 1 sh test-new.sh 2>&1", cwd=CWD, shell=True,
```
Also, why `test-new.sh`?

nmdis1999 replied:
I didn't mean to commit this change, it was purely from testing purpose. Will revert this.

vext01 commented on ykrt/yklua_tests.sh:
```diff
@@ -0,0 +1,33 @@
+cd tests
+
+LUA=../src/lua
+# XXX we should repeat, say, 5 or 10 times to try to rule out non-deterministic
```
Since there is repetition in the script that calls this one, do we need repetition here?

nmdis1999 commented:
Had to merge master before addressing your comment @vext01 , [45543f1](https://github.com/ykjit/yk/pull/854/commits/45543f1c18368b2e3b73be341362b0d333f77fe5) consist of the changes you asked and I moved `yklua` test script to `pass-finder` folder under `ykrt`.

vext01 commented:
The change looks OK I think, but it was harder than necessary to verify because you force pushed.

Please only force push when the reviewer is OK with it.

If you needed to merge master, then you can stack a merge commit on the old commits and then add any extra changes after that. This way I can see exactly what changed without having the changes mixed in with earlier changes.

Anyway, this one is now looking good. Please squash your commits.

nmdis1999 commented:
Squashed [b7e6f9a](https://github.com/ykjit/yk/pull/854/commits/b7e6f9acdfe0257409f84eb66056bcc3cc5f4e06).

vext01 commented:
bors r+
